docs/dev/Metadata.md

```diff
@@ -0,0 +1,136 @@
+# Metadata
+
+Metadata is a JavaScript object, containing information about the public interface of a UI5 Web Component (tag name, properties, etc...)
+
+*Disclaimer: The information below is for UI5 Web Components development only. It is experimental and may change*
+
+## Tag
+
+Defines the HTML tag for the Web Component.
+
+#### Example:
+
+```json
+{
+	"tag": "ui5-my-element",
+}
+``` 
+
+## Properties / Attributes
+
+Defines the HTML properties for the Web Component. 
+
+*Note about attributes:* By default, for each property an equivalent attribute is supported. Attributes have the same names as properties, but in `kebab-case` rather than `camelCase`.
+Properties of type `Object`, properties with `multiple` set to`true` and properties with `noAttribute` set to `true` do not have an attribute equivalent. 
+
+#### Example
+
+```json
+{
+	"properties": {
+		"message": {
+			"type": String,
+			"defaultValue": "Hello",
+		},
+		"shown": {
+			"type": Boolean,
+			"noAttribute": true,
+		},
+		"settings": {
+			"type": Object,
+			"deepEqual": true,
+		},
+		"nums": {
+			"type": Integer,
+			"multiple": true,
+		},
+		"animationDuration": {
+			"type": Integer,
+			"nonVisual": true,
+		},
+		"width": {
+			"type": CSSSize,
+			"defaultValue": "",
+		},
+	},
+}
+```
+
+#### Property configuration settings
+
+Setting | Type | Default | Description
+--------|------|--------| -----------
+`type`    | Property type | N/A | The type of the property. For more information on types see the table below. 
+`defaultValue` | Any valid value for the type | undefined | Default value of the property. Cannot be set for type "Boolean". Booleans are always false by default in HTML
+`multiple` | Boolean | false | Indicates whether the property represents a single value or is an array of values of the given type
+`noAttribute` | Boolean | false | No attribute equivalent will be created for that property. Always false for properties of type Object.
+`deepEqual` | Boolean | false | Deep equal object comparison, rather than reference comparison, will be used to determine if the value of the property changed.
+`nonVisual` | Boolean | false | Changes to this property will not lead to invalidation of the Web Component. Use for properties, not related to rendering.
```

Comment:
`nonVisual` is no longer needed, we can remove it, used only in `DayPicker.js` and `DayPicker.hbs`

Reply:
let`s discuss the `noAttribute`, we may remove this one as well.

Reply:
`multiple` can be replaced by type: `Array` or even `Object`

Reply:
let`s remove `deepEqual` and  either check deep always or better just react on reference re-assignment as guideline with Object.assign({ x }, y);